app/actions/websocket/index.ts

```diff
@@ -124,13 +124,14 @@ async function doReconnect(serverUrl: string) {
     }
     const {models, initialTeamId, initialChannelId, prefData, teamData, chData} = entryData;
 
-    await handleEntryAfterLoadNavigation(serverUrl, teamData.memberships || [], chData?.memberships || [], currentTeamId || '', currentChannelId || '', initialTeamId, initialChannelId);
-
     const dt = Date.now();
-    await operator.batchRecords(models, 'doReconnect');
+    if (models?.length) {
+        await operator.batchRecords(models, 'doReconnect');
+    }
     logInfo('WEBSOCKET RECONNECT MODELS BATCHING TOOK', `${Date.now() - dt}ms`);
     setTeamLoading(serverUrl, false);
 
+    await handleEntryAfterLoadNavigation(serverUrl, teamData.memberships || [], chData?.memberships || [], currentTeamId || '', currentChannelId || '', initialTeamId, initialChannelId);
```

Comment:
If we kick them after the models are batched, we may have two problems:
- The name of the channel that we are kicking the user out from is no longer available
- The channel / thread / team logic regarding "this no longer exist, I will move somewhere else" may kick in.

What exactly are you trying to solve moving this here?

Reply:
we can move it up if you feel strongly about it, it was just it makes sense to execute after the batch gets executed.

Reply:
Yes, at the beginning I thought the same, but then the errors just told you about came to bite my back :sweat_smile: 
Making this before the batch can lead to some errors on corner cases (while away, kicked from team A and added to team B, you may try to navigate to team B and it will fail, I think), but those are going to happen rarely, and we can solve that in the `handleEntryAfterLoadNavigation` function.